conan/tools/google/bazeldeps.py

```diff
@@ -115,41 +118,69 @@ def _relativize_path(p, base_path):
         headers = ', '.join(headers)
         includes = ', '.join(includes)
 
-        defines = ('"{}"'.format(define.replace('"', "'"))
+        defines = ('"{}"'.format(define.replace('"', '\\' * 3 + '"'))
                    for define in cpp_info.defines)
         defines = ', '.join(defines)
 
         linkopts = []
-        for linkopt in cpp_info.system_libs:
-            linkopts.append('"-l{}"'.format(linkopt))
-        linkopts = ', '.join(linkopts)
-
+        for system_lib in cpp_info.system_libs:
+            # FIXME: Replace with settings_build
+            if platform.system() == "Windows":
+                linkopts.append('"/DEFAULTLIB:{}"'.format(system_lib))
+            else:
+                linkopts.append('"-l{}"'.format(system_lib))
 
+        linkopts = ', '.join(linkopts)
         lib_dir = 'lib'
-        if len(cpp_info.libdirs) != 0:
-            lib_dir = _relativize_path(cpp_info.libdirs[0], package_folder)
 
         dependencies = []
         for req, dep in dependency.dependencies.items():
             dependencies.append(dep.ref.name)
 
+        libs = {}
+        for lib in cpp_info.libs:
+            real_path = self._resolve_real_file(cpp_info.libdirs, lib)
+            if real_path:
+                libs[lib] = _relativize_path(real_path, package_folder)
+
         shared_library = dependency.options.get_safe("shared") if dependency.options else False
         context = {
             "name": dependency.ref.name,
-            "libs": cpp_info.libs,
+            "libs": libs,
             "libdir": lib_dir,
             "headers": headers,
             "includes": includes,
             "defines": defines,
             "linkopts": linkopts,
             "library_type": "shared_library" if shared_library else "static_library",
-            "extension": "so" if shared_library else "a",
             "dependencies": dependencies,
         }
         content = Template(template).render(**context)
         return content
 
+    def _resolve_real_file(self, libdirs, lib):
+        for libdir in libdirs:
```

Comment:
Note that this is looking for the real files (code extracted from the collect_libs)

Reply:
Renamed to `_get_libs_absolute_paths`. Is that better?

Reply:
Fine for me 😁

Reply:
This could be close to the code that we would want to do to detect the static/shared libs, in order to avoid the UNKNOWN IMPORTED in CMakeDeps. Might be worth refactorizing from here (later, I am commenting here for awareness) if that is the way forward.

Reply:
Yep, I thought the same.